orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionRunnerSupport.java

```diff
@@ -113,13 +105,29 @@ public ExecutionRunnerSupport(Collection<StageDefinitionBuilder> stageDefinition
         branchBuilder.postBranchGraph(stage, postBranch);
       });
       callback.accept(singleton(stage), afterGraph);
+
+      // ensure parallel stages have the correct stage type (ie. createServerGroup -> deploy to satisfy deck)
+      parallelStages.forEach(it -> it.setType(branchBuilder.getType()));
     } else {
+      Map<SyntheticStageOwner, List<Stage<T>>> aroundStages = builder
+        .aroundStages(stage)
+        .stream()
+        .collect(groupingBy(Stage::getSyntheticStageOwner));
+
+      if (stage.getExecution().getBuiltPipelineObjects().contains(stage)) {
+        aroundStages = new HashMap<>();
```

Comment:
The execution bit was something we were doing on the old model ... the `ExecutionRunnerSupport` is a component so it's life cycle doesn't really jive with per-execution/